### ISSUE TITLE
In keyword-spacing overrides option, we should be able to disable a check at all

### ISSUE THREAD

**Issue by julienw** [eslint/eslint#5336]
Currently, `before`, `after` and `overrides` entries are boolean to allow or disallow the spacing. But it would be useful to be able to just disable the check: for some cases, we don't want to check the spacing.

For example in my styleguide I don't mind whether `switch` is followed by a space or not.

I could configure like this:
```javascript
"keyword-spacing": [2, { "overrides": {
  "switch": { "after": null /* or "no-check" ? */ }
}}],
```

**Comment by eslintbot** [comment]
Hi @julienw, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by julienw** [comment]
I use eslint v2.1.0.

Code examples are
```
switch(variable) {
  case 'value':
    break;
}

switch (variable) {
  case 'value':
    break;
}
```

I'd like to use `keyword-spacing` to check spacing for other keywords (like `if`, `for` etc), but I don't want to check for `switch`, which means that both cases above should not give an error.

So I want to be able to configure the `keyword-spacing` rule to disable one of the keyword check. Now that I think this a little more, it should be possible to chose if it's a warning/error as well.

The configuration could then be:
```javascript
"keyword-spacing": [2, { "overrides": {
  "switch": { "after": { "severity": 0/1/2, "has-space": true/false }
}}],
```

**Comment by ilyavolodin** [comment]
Rule severity is controlled by the core engines. Rules can't affect it. So severity is not possible to change. Adding `overrides` for different keywords seems like an overkill to me, but let's see what the rest of the team feels like.
CC: @eslint/eslint-team

**Comment by nzakas** [comment]
We already have `overrides` in that rule: http://eslint.org/docs/rules/keyword-spacing#options

What he's asking for is another way to configure the overrides so as to say "don't check spacing for this keyword".

The only way we can possibly do this is something like:

```
"keyword-spacing": [2, { "overrides": {
  "switch": "ignore"
}}],
```

I'm not sure if it's a good idea to do so, though.

**Comment by BYK** [comment]
I think we can use the `mode: "strict"` and `mode: "minimum"` distinction from `key-spacing` rule here.

**Comment by nzakas** [comment]
@byk can you explain how that would work?

**Comment by BYK** [comment]
How about:

```json
"keyword-spacing": [2, { "overrides": {
  "switch": { "mode": "minimum"}
}}],
```
This also implies adding a "global" `mode` option.

**Comment by nzakas** [comment]
Hmm, in key-spacing "minimum" means any number of spaces is valid. What's being requested here seems to be "disable", as in, don't check at all.

**Comment by BYK** [comment]
"minimum" combined with before/after: `false` does exactly that though, right?

**Comment by nzakas** [comment]
@byk that's a long trip to get to the actual meaning. I don't think it's clear that's what would happen.

**Comment by BYK** [comment]
@nzakas yeah, I agree :(

**Comment by julienw** [comment]
What makes things difficult here is that the `keyword-spacing` rule handles all keywords.. It would be easier if there were separate rules for separate keywords.

**Comment by nzakas** [comment]
@julienw that would be a nightmare for end users. That level of granularity is overwhelming for people when they just want one or two exceptions.

**Comment by julienw** [comment]
@nzakas Maybe we'd need separate rules but also an alias that would represent several separate rules; but it's a long shot just to fix this :)

**Comment by nzakas** [comment]
@julienw we don't have a way to do that, and that seems like overkill for this problem.

**Comment by julienw** [comment]
Yes. I don't have better ideas than my first comment https://github.com/eslint/eslint/issues/5336#issue-134830753 or your first comment https://github.com/eslint/eslint/issues/5336#issuecomment-189921463

**Comment by nzakas** [comment]
@eslint/eslint-team does anyone want to champion this?

**Comment by platinumazure** [comment]
I would love to take a crack at proposing a schema for this, if anyone from the team is willing to champion it.

**Comment by nzakas** [comment]
@platinumazure are you still interested in championing this? If not, then it's time to close this.

**Comment by platinumazure** [comment]
@nzakas I'm still interested, but this fell off my radar and I won't be able to draw up a schema until maybe this weekend. I'll close for now. Sorry for holding things up.
